### DISCUSSION TITLE
How to set form state if obj shape is same?

### DISCUSSION THREAD

**Discussion by popuguytheparrot** [react-hook-form/react-hook-form#7662] [Q&A] [answered]
How to set form state if obj shape is same without duplicate each field in setValues array?

**Comment by Moshyfawn** [comment]
Do u have an example of your state object?

**Reply by popuguytheparrot** [reply]
{id, name, type}

i just want do smth like that

fetch(id).then(setFormState)

**Reply by popuguytheparrot** [reply]
without `reset(Object.keys(response).map(key => ({[key]: response[key]})))`

**Comment by Moshyfawn** [comment, marked as answer]
Have you looked at [the `reset` form method](https://react-hook-form.com/ru/api/#reset)?
```js
const { reset } = useForm()

const getRemoteData = useCallback(async () => {
  const fieldValues = await fetch(`https://such-api/much-data/${itemId}`)
  reset(fieldValues)
  // или
  reset({
    name: fieldValues.nameVar
  })
}, [reset])

useEffect(() => {
  getRemoteData()
}, [getRemoteData])
```

**Reply by popuguytheparrot** [reply]
Ok i try, but naming for this method is weird

**Reply by Moshyfawn** [reply]
RHF expects to have `defaultValues` passed at the hook level
```js
useForm({
  defaultValues: {
    name: ""
  }
})
```
Even if you don't pass `defaultValues`, the form itself has already been initialized with `undefined` for `defaultValues`, so what you do is you **reset** form's values with your remote data you get asynchronously after it's initialization

**Reply by popuguytheparrot** [reply]
i understand, but form data sets after request
